pkg/lib/app_detection.go

```diff
@@ -0,0 +1,418 @@
+			dt.Logger.Info("Web Application Keptn " + project + " " + stage + " already exists.")
+			return item.ID, nil
+		}
+	}
+
+	dt.Logger.Info("Creating new Web Application: Keptn " + project + " " + stage)
+	payload, err := json.Marshal(application)
+
+	if err != nil {
+		return "", err
+	}
+	resp, err = dt.sendDynatraceAPIRequest("/api/config/v1/applications/web", "POST", string(payload))
+
+	if err != nil {
+		return "", err
+	}
+
+	responseItem := &Values{}
+
+	err = json.Unmarshal([]byte(resp), responseItem)
+	if err != nil {
+		return "", err
+	}
+
+	dt.Logger.Info("Successfully created Web Application: Keptn " + project + " " + stage + ". ID=" + responseItem.ID)
+	return responseItem.ID, nil
+
+}
+
+func (dt *DynatraceHelper) createAppDetectionRule(project string, stage string, webAppID string) (string, error) {
+	detectionRule := createAppDetectionRule(project, stage, webAppID)
+	dt.Logger.Info("Checking for existing Application Detection Rules for Keptn " + project + " " + stage)
+	resp, err := dt.sendDynatraceAPIRequest("/api/config/v1/applicationDetectionRules", "GET", "")
+	if err != nil {
+		return "", err
+	}
+
+	items := &DTAPIListResponse{}
+
+	err = json.Unmarshal([]byte(resp), items)
+	if err != nil {
+		return "", err
+	}
+
+	for _, item := range items.Values {
+		if strings.ToLower(item.Name) == strings.ToLower(getWebApplicationName(project, stage)) {
+			dt.Logger.Info("Web Application Detection Rule for Keptn " + project + " " + stage + " already exists.")
+			return item.ID, nil
+		}
+	}
+
+	dt.Logger.Info("Creating new Application Detection Rule for Keptn " + project + " " + stage)
+	payload, err := json.Marshal(detectionRule)
+
+	if err != nil {
+		return "", err
+	}
+	resp, err = dt.sendDynatraceAPIRequest("/api/config/v1/applicationDetectionRules", "POST", string(payload))
+
+	if err != nil {
+		return "", err
+	}
+
+	responseItem := &Values{}
+
+	err = json.Unmarshal([]byte(resp), responseItem)
+	if err != nil {
+		return "", err
+	}
+
+	dt.Logger.Info("Successfully created Application Detection Rule for Keptn " + project + " " + stage + ". ID=" + responseItem.ID)
+	return responseItem.ID, nil
+}
+
+func createAppDetectionRule(project string, stage string, webAppID string) *AppDetectionRule {
+	return &AppDetectionRule{
+		ApplicationIdentifier: webAppID,
+		FilterConfig: ADRFilterConfig{
+			Pattern:                project + "-" + stage,
+			ApplicationMatchType:   "CONTAINS",
+			ApplicationMatchTarget: "DOMAIN",
+		},
+	}
+}
+
+func createWebApplication(project string, stage string) *WebApplication {
+	return &WebApplication{
+		Name:                             getWebApplicationName(project, stage),
+		Type:                             "AUTO_INJECTED",
+		RealUserMonitoringEnabled:        true,
+		CostControlUserSessionPercentage: 100.0,
+		LoadActionKeyPerformanceMetric:   "VISUALLY_COMPLETE",
+		XhrActionKeyPerformanceMetric:    "VISUALLY_COMPLETE",
+		LoadActionApdexSettings: WALoadActionApdexSettings{
```

Comment:
That is a lot of settings that we need to maintain for a simple application.

I want to avoid that we have all those "default values" in our code, when this is clearly something that should be handled within the Dynatrace Api. 

Please get some feedback from the Api Developers with respect to simplification of this Api call.